sdk/core/Azure.Core/src/DelayStrategy.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// Strategy to control delay behavior.
+    /// </summary>
+    public abstract class DelayStrategy
+    {
+        private const string RetryAfterHeaderName = "Retry-After";
+        private const string RetryAfterMsHeaderName = "retry-after-ms";
+        private const string XRetryAfterMsHeaderName = "x-ms-retry-after-ms";
+        /// <summary>
+        /// Get the interval of next delay iteration.
+        /// </summary>
+        /// <remarks> Note that the value could change per call. </remarks>
+        /// <param name="response"> Server response. </param>
+        /// <param name="attempt"></param>
+        /// <param name="suggestedDelay"> Suggested pollingInterval. It is up to strategy
+        /// implementation to decide how to deal with this parameter. </param>
+        /// <returns> Delay interval of next iteration. </returns>
+        public abstract TimeSpan GetNextDelay(Response response, int attempt, TimeSpan? suggestedDelay);
+
+        /// <summary>
+        /// Gets the server specified delay. If the message has no response, <see cref="TimeSpan.Zero"/> is returned.
+        /// This method can be used to help calculate the next delay when overriding <see cref="GetNextDelay"/>, i.e.
+        /// implementors may want to add the server delay to their own custom delay.
+        /// </summary>
+        /// <param name="message">The message to inspect for the server specified delay.</param>
+        /// <returns>The server specified delay.</returns>
+        protected internal static TimeSpan GetServerDelay(HttpMessage message)
```

Comment:
This should probably take in the Response for consistency.

Reply:
Though I suppose that would mean all callers would need to do the HasResponse check themselves so maybe it is okay to leave this as HttpMessage.

Reply:
Methods in our public APIs that take HttpMessage usually do so because we want to be able to pull values from either the request or the response.  Since this depends on a value from the server exclusively, i.e. a response, it would probably make sense to scope it directly to that.